File: Java/src/BullsandCows.java

public class BullsandCows {
    
    public String getHint(String secret, String guess) {
        int bulls = 0;
        int cows = 0;
        
        int [] count = new int[10];
        
        for(int i = 0; i < secret.length(); i++) {
            if(secret.charAt(i) == guess.charAt(i)) bulls++;
            else count[secret.charAt(i) - '0']++;
        }
        
        for(int i = 0; i < secret.length(); i++) {
            
            if(secret.charAt(i) != guess.charAt(i)) {
                if(count[guess.charAt(i) - '0'] > 0) {
                    cows++;
                    count[guess.charAt(i) - '0']--;
                }
            }
            
        }
        String result = "" + bulls + "A" + cows + "B";
        return result;
    }
    
}

File: Python/BullsandCows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        n = len(secret)
        bulls = 0
        count = {}
        for i in range(n):
            if secret[i] == guess[i]:
                bulls += 1
            else:
                if secret[i] in count:
                    count[secret[i]] += 1
                else:
                    count[secret[i]] = 1
        cows = 0
        for i in range(n):
            if secret[i] != guess[i]:
                if guess[i] in count and count[guess[i]] > 0:
                    cows += 1
                    count[guess[i]] -= 1
        return str(bulls) + "A" + str(cows) + "B"



test = Solution()
secret = "1123"
guess = "0111"
print(test.getHint(secret, guess))